config/templates/keepalived-template.yaml

```diff
@@ -12,6 +12,10 @@
         keepalivedGroup: {{ .KeepalivedGroup.ObjectMeta.Name }}
     template:
       metadata:
+        annotations:
+        {{ range $index, $element := .KeepalivedGroup.Spec.DaemonsetAnnotations }}
```

Comment:
did you test it? I think this is going to fail if there are no annotations. I suggest you take inspiration from the helm default behavior. I.e. when you create a new chart.

Reply:
Seems more concise this way since it happens to work but let me know if you want me to rework it anyway. I'd be happy to do so if you think it's worth a null check